### PR TITLE
Bitpanda

### LINKED ISSUE #1146: FR: support Bitpanda exchange
Selfishly, this would be really helpful to me ;-)  But it's a major exchange so it would benefit other users too.

### PR DESCRIPTION
**Pull request by LefterisJP**
Closes #1146

Frontend still needs to be done

### CODE REVIEW

File: rotkehlchen/exchanges/bitpanda.py [hunk 1]
```diff
@@ -0,0 +1,537 @@
...
+            from_ts: Optional[Timestamp],
+            to_ts: Optional[Timestamp],
+            options: Optional[Dict[str, Any]] = None,
+    ) -> Union[List[Trade], List[AssetMovement]]:
+        """Query a paginated endpoint until all pages are read
+
+        May raise RemoteError
+        """
+        from_timestamp = from_ts if from_ts is not None else Timestamp(0)
+        to_timestamp = to_ts if to_ts is not None else ts_now()
```
Comment (yabirgb) [inline rotkehlchen/exchanges/bitpanda.py:402]: I don't know if this needs to be checked here but should we check that `to_timestamp > from_timestamp`?

**Review by yabirgb — COMMENTED** [review]
Two small comments. Everything else looks good to me :)

File: rotkehlchen/exchanges/bitpanda.py [hunk 2]
```diff
@@ -0,0 +1,537 @@
...
+    ):
+        super().__init__(
+            name=name,
+            location=Location.BITPANDA,
+            api_key=api_key,
+            secret=secret,
+            database=database,
+        )
+
+        # self.uri = 'https://api.exchange.bitpanda.com/public/v1'
```
[the remark this answers is not in the document]
Comment (LefterisJP) [reply rotkehlchen/exchanges/bitpanda.py:66]: AAAAA ffs thanks for finding it

File: rotkehlchen/exchanges/bitpanda.py (hunk 1 above)
Comment (LefterisJP) [reply rotkehlchen/exchanges/bitpanda.py:402]: We don't do that anywhere where we use from_ts and to_ts. If at any point we do it, it should be in the marshmallow checker imo

File: rotkehlchen/exchanges/coinbasepro.py [hunk 3]
```diff
@@ -235,6 +235,7 @@ def _api_query(
                     request_method.lower(),
                     full_url,
                     data=stringified_options,
+                    timeout=DEFAULT_TIMEOUT_TUPLE,
```
Comment (LefterisJP) [inline rotkehlchen/exchanges/coinbasepro.py:238]: hah this caused all coinbase tests to fail due to mocking. Lol. Let me fix.

**Comment by LefterisJP** [comment]
So all is good. Test passes locally and was just an etherscan timeout. The lgtm is a false positive which is commented in the code but neeeds to be committed (if I understand correctly) for lgtm to shut up

**Pull request merged**